### ISSUE TITLE
Executing "close" function doesn't close session

### ISSUE THREAD

**Issue by choxnox** [Wisembly/elephant.io#97]
Due to simplicity I've decided to throw out irrelevant parts of code. Running the following code throws exception on the second `try-catch` block:
```
$hostname = 'my.host.name';
$port = 'myport';

$client = new Client(new Version1X(sprintf("https://%s:%s", $hostname, $port)), new Logger());

try
{
    $client->initialize();
    $client->emit('sendMail', $mailData);
    $client->close();
} 
catch (Exception $e) 
{
    error_log($e);
}

try
{
    $client->initialize();
    $client->emit('sendMail', $mailData);
    $client->close();
} 
catch (Exception $e) 
{
    error_log($e);
}
```

Class `Logger` simply prints out messages to PHP's `error_log`. I get this output in the error log:

```
[08-Mar-2015 20:54:01 Europe/Berlin] Connecting to the websocket
[08-Mar-2015 20:54:01 Europe/Berlin] Connected to the server
[08-Mar-2015 20:54:01 Europe/Berlin] Sending a new message
[08-Mar-2015 20:54:01 Europe/Berlin] Closing the connection to the websocket
[08-Mar-2015 20:54:01 Europe/Berlin] Connecting to the websocket
[08-Mar-2015 20:54:01 Europe/Berlin] exception 'UnexpectedValueException' with message 'The server returned an unexpected value. Expected "HTTP/1.1 101", had ""'
```

Needless to say, my second emitted message never reaches Node.js server. I did some Google researching and digging into Elephant.io code and after chasing the wild goose for few hours I've managed to pinpoint the issue.

I've resolved the problem by deleting session variable inside `Version1X` engine when `close()` function is called. Basically what I did was add the following line at the end of `close()` function inside `Version1X.php` file:

```
public function close()
{
    ...
    $this->session = null;
}
```

Since I'm new to both Node.js and Socket.io (but not to the PHP/JS development) I'm not sure if session should be killed when connection is closed? When `connect()` function is called it calls `handshake()` function which in turn doesn't execute if session is already existing.

If I recreate the `Client` before the second `try-catch` block then it works OK.

**Comment by Taluu** [comment]
Makes sense. Could you make a PR ? Thanks

**Comment by Taluu** [comment]
Fixed by #108
